LanguageFeatures/Static-access-shorthand/equality_A02_t05.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion For `==`, we special-case when the right operand is (precisely!)
+/// a static member shorthand.
+/// ...
+/// This special-casing is only against an immediate static member shorthand. It
+/// does not change the context type of the second operand, so it would not work
+/// with, for example, `Endian.host == wantBig ? .big : .little`. Here the
+/// second operand is not a `<staticMemberShorthand>`, so it won't have a
+/// shorthand context set, and the context type of the second operand of `==` is
+/// the empty context `_`. (It's neither the static type of the first operand,
+/// nor the parameter type of the first operand's `operator==`.)
+///
+/// @description Checks that it is a compile-time error if a pattern has the
+/// form `== e2` or `!= e2` and `e2` is not a precisely shorthand expression.
+/// Test a class.
```

Comment:
But these are not patterns, they are equality expressions. Looks like the description or the test should be adjusted for consistency.

Reply:
Ah, no. The idea was to have patterns. Here and below added `case`. Thank you for noticing!